packages/monaco/src/browser/monaco-editor-model.ts

```diff
@@ -52,8 +54,10 @@ export class MonacoEditorModel implements ITextEditorModel, TextEditorDocument {
     protected readonly onDidSaveModelEmitter = new Emitter<monaco.editor.IModel>();
     readonly onDidSaveModel = this.onDidSaveModelEmitter.event;
 
-    protected readonly onWillSaveModelEmitter = new Emitter<WillSaveMonacoModelEvent>();
-    readonly onWillSaveModel = this.onWillSaveModelEmitter.event;
+    protected readonly onWillSaveModelListeners: WillSaveMonacoModelListener[] = [];
+    onWillSaveModel(listener: WillSaveMonacoModelListener) {
```

Comment:
`onWillSaveModel` should have `Event<WillSaveMonacoModelEvent>` type as in vscode API